packages/content/parsers/markdown/compilers/json.js

```diff
@@ -13,10 +15,14 @@ function parseAsJSON (node, parent) {
     /**
      * Replace a tag with nuxt-link if relative
      */
-    if (node.tagName === 'a' && (node.properties.href || '').startsWith('/')) {
-      node.tagName = 'nuxt-link'
-      node.properties.to = node.properties.href
-      delete node.properties.href
+    if (node.tagName === 'a') {
+      const pathname = (node.properties.href || '')
+
+      if (pathname.startsWith('/') && extname(pathname) === '') {
```

Comment:
Using `extname` for detect static files is not safe. For example returning value of `extname` for url `/setup#install-nuxt.js` is `.js`, which isn't correct and leads to invalid behavior.

Static detection should be done in markdown parser and ensure the file exists in `static` directory

Reply:
I'm confused about what you mean it should be done in the markdown parser. This is the markdown to json logic, and where it does the link modification...

As for using extname, it seemed like the fastest of two not great options. I can replace this with a `fs.existsSync` call instead but that would probably slow down the build time for large sites. It will be ran for every absolute link. Alternatively, I can strip out the hash from the path name, and still do the same extname logic.

Reply:
Without checking file existence, this will be a breaking change. Since pages that have '.' in their names, will force full reload instead of client side navigation. 

As for the performance it can be improved by prefetching static files list.

Reply:
Yep!

Sadly it would require us to get the nuxt options to that function so we can use the configured static folder. Looking at the code, that shouldn't be too hard. Then we'd just need to watch the static folder for changes.

Considering the amount of work for that change, I'm going to hold off on writing it til I can get a Nuxt team member to comment on what they would like, as they are going to be the ones maintaining it.